core.c

```diff
@@ -336,29 +330,28 @@ void core_reset() {
 /**
  * @name	core_check_gl_error
  * @brief	check's for a gl error and reports it to the platform
- * @retval	NONE
+ * @retval	(bool) whether or not there was an error
  */
-void core_check_gl_error() {
+bool core_check_gl_error() {
```

Comment:
This is a good suggestion - I don't have time to refactor yet though - because there are a lot of similar changes I would make. These fixes focus on working logic so we can ship safely on all devices. Let's circle back and optimize / refactor when we have time!